### PR TITLE
dot/network: update DHT to work with kusama, increase peer count

### LINKED ISSUE #1134: Sync Kusama Network
## Expected Behavior

- Connect to Kusama network and sync to the head

## Current Behavior

- N/A

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [ ] I am planning to submit a pull request to fix this issue myself

### LINKED ISSUE #1331: syncing: continuously receive block response messages on same stream
## Task summary

- there is currently an issue when syncing where we can receive the first block response fine, but after that, we get a "stream reset" error after sending the second request and trying to receive the response
- investigate and fix this, maybe the peer is resetting the stream for some reason?

## Other information and links

-  related to #1134

### LINKED ISSUE #1346: get DHT discovery working with kusama
## Task summary

- currently, when the node connects to a kusama node, fails to begin advertising in the DHT: `failed to begin advertising: failed to find any peer in table`
- this may be because we begin advertising before being connected to any peers, or perhaps the DHT approach we are using is wrong
- fix usage of the DHT so that we can discover new peers via the DHT

### LINKED ISSUE #1364: add min peers, max peers to network config
## Task summary

- we should be able to configure minimum and maximum peers from the config file for use in the ConnManager
- add `min-peers` and `max-peers` fields under `[network]` in the config.toml

### PR DESCRIPTION
**Pull request by noot**
## Changes

- update DHT to work with kusama (bump version, update protocol ID)
- increase max peer count to 50 
- syncing now works fully, we have lots of peers :D

## Tests

```
go test ./dot/network -short
```

as usual please run instructions in #1312 (after node has peers, you can turn off local substrate kusama node)

note: with these updates, it takes a bit longer to start to sync, because it waits for a block announce message from some peer before sending the block request (the local kusama node no longer opens the stream with us immediately) so you will have to let the node run for a minute or so, then you will see if begin to sync.

also, turn on network debug logs in the config to see the peer count and extra info

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [ ] All integration tests and required coverage checks are passing

## Issues

- closes #1346
- closes #1331
- closes #1364
- related to #1134

### CODE REVIEW

**Review by edwardmack — APPROVED** [review]
This looks good, however syncing stops for me with this error:
```
INFO[02-08|17:02:06] imported block                           pkg=sync number=256 hash=0xe3eb7632fe5e2ef3560278d0ba76dc7ae0552058e591a82d9f029d758a8a6d6b caller=syncer.go:468
EROR[02-08|17:02:06] Failed to read LEB128 encoding           pkg=network protocol=/ksmcc3/sync/2 error="stream reset" caller=service.go:426
EROR[02-08|17:02:06] Failed to read LEB128 encoding           pkg=network protocol=/ksmcc3/transactions/1 error="stream reset" caller=service.go:426
EROR[02-08|17:02:06] Failed to read LEB128 encoding           pkg=network protocol=/ksmcc3/block-announces/1 error="stream reset" caller=service.go:426
```

**Comment by noot** [comment]
@edwardmack i'll update that, it just means a peer disconnected from us. as long as we have other peers it should be ok

File: dot/network/host.go [hunk 1]
```diff
@@ -84,7 +83,7 @@ func newHost(ctx context.Context, cfg *Config) (*host, error) {
 		libp2p.Identity(cfg.privateKey),
 		libp2p.NATPortMap(),
 		libp2p.ConnectionManager(cm),
-		libp2p.ChainOptions(libp2p.DefaultSecurity, libp2p.Security(noise.ID, noise.New)),
+		libp2p.ChainOptions(libp2p.DefaultSecurity, libp2p.Security(secio.ID, secio.New)), // TODO: deprecate secio?
```
Comment (RyRy79261) [inline dot/network/host.go:86]: Should this be a ticket perhaps? Could add the issue number if so

**Review by RyRy79261 — APPROVED** [review]
Looks good, not able to run the node to run tests against but just to mention, there's also an empty `test` file, not sure if thats needed

**Comment by edwardmack** [comment]
> @edwardmack i'll update that, it just means a peer disconnected from us. as long as we have other peers it should be ok
> 
> update: those logs were already changed to debug, can you `make gossamer` on this branch and try again?

It's running without error now, I must have been running with stale binary, sorry about that.  Note, it's now averaging about 2 blocks a seconds, much less than I was getting before (around 8 blocks a second).

**Comment by noot** [comment]
@edwardmack the speed improvements are in my other branch, this is only for the DHT. when they are both merged it should be fast still

**Pull request merged**